src/main/kotlin/sh/hsp/techtree/graphviz/DSLConverter.kt

```diff
@@ -0,0 +1,26 @@
+package sh.hsp.techtree.graphviz
+
+import sh.hsp.techtree.TreeModel
+import sh.hsp.techtree.TreeNode
+import java.util.stream.Collectors
+
+class DSLConverter {
+    fun convert(model: TreeModel): String {
+         val nodesAsDSL = model.nodes.stream()
+            .map { registerNode(it) }
+            .collect(Collectors.joining(" "))
+
+        return "digraph { $nodesAsDSL }"
+    }
+
+    private fun registerNode(node: TreeNode): String {
```

Comment:
Pa na to:
```
    private fun registerNode(node: TreeNode): String =
        """
        ${node.title}
        ${
            (node.requires ?: emptyList())
                .stream()
                .map { "${node.title} -> $it" }
                .collect(Collectors.joining("\n"))
        }
        """.trimIndent()
```
Tu niestety się identy posypią, ale zostawię ćwiczenie dla czytelnika.

Reply:
:hmm: